src/ShellClients/HideTracker.vala

```diff
@@ -196,23 +215,44 @@ public class Gala.HideTracker : Object {
 #endif
 
         if (should_hide && !hovered && !panel.window.has_focus ()) {
-            // Don't hide if we have transients, e.g. an open popover, dialog, etc.
-            var has_transients = false;
-            panel.window.foreach_transient (() => {
-                has_transients = true;
-                return false;
-            });
-
-            if (has_transients) {
-                return;
-            }
+            trigger_hide ();
+        } else {
+            trigger_show ();
+        }
+    }
 
+    private void trigger_hide () {
+        // Don't hide if we have transients, e.g. an open popover, dialog, etc.
+        var has_transients = false;
+        panel.window.foreach_transient (() => {
+            has_transients = true;
+            return false;
+        });
+
+        if (has_transients) {
+            reset_hide_timeout ();
+
+            return;
+        }
+
+        hide_timeout_id = Timeout.add_once (HIDE_DELAY, () => {
             hide ();
-        } else {
-            show ();
+            hide_timeout_id = 0;
+        });
+    }
+
+    private void reset_hide_timeout () {
```

Comment:
This should also be called in the destructor of this class or otherwise we might get a crash when e.g. revealing the dock to open settings and then switching to NEVER quick enough.